Runtime/src/index.ts

```diff
@@ -420,6 +434,29 @@ export class SwiftRuntime {
                 });
                 writeUint32(func_ref_ptr, func_ref);
             },
+            swjs_call_throwing_new: (
+                ref: ref,
+                argv: pointer,
+                argc: number,
+                result_obj: pointer,
+                exception_kind_ptr: pointer,
+                exception_payload1_ptr: pointer,
+                exception_payload2_ptr: pointer
+            ) => {
+                const obj = this.heap.referenceHeap(ref);
+                let result: any
+                try {
+                    result = Reflect.construct(obj, decodeValues(argv, argc));
+                    if (typeof result != "object")
```

Comment:
I’m pretty sure `new` can’t return a non-object. If you try to return a non-object in a constructor, it will use the `this` value instead.

See the spec [here](https://read262.jedfox.com/ordinary-and-exotic-objects-behaviours/ecmascript-function-objects#sec-ecmascript-function-objects-construct-argumentslist-newtarget), where step 12 is the most relevant.